### PR TITLE
Update: add end location to report in `consistent-return` (refs #12334)

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Other, please explain:

refs #12334

This PR adds `loc.end` to reports in `consistent-return`

#### What changes did you make? (Give an overview)

Just a small change from `loc.start` to `loc`

#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

File: tests/lib/rules/consistent-return.js [hunk 1]
```diff
@@ -255,6 +316,7 @@ ruleTester.run("consistent-return", rule, {
                     messageId: "missingReturn",
                     data: { name: "program" },
                     type: "Program",
+                    line: 1,
                     column: 1
```
Comment (btmills) [inline tests/lib/rules/consistent-return.js:320]: Every other invalid test case asserts end location information, but we don't have an end location for `Program` reports. Do you think it's worth adding these to demonstrate to future readers that this omission is intentional?

```suggestion
                    line: 1,
                    column: 1,
                    endLine: void 0,
                    endColumn: void 0 
```

File: tests/lib/rules/consistent-return.js [hunk 2]
```diff
@@ -207,7 +256,10 @@ ruleTester.run("consistent-return", rule, {
                     messageId: "missingReturn",
                     data: { name: "function" },
                     type: "FunctionExpression",
-                    column: 3
+                    line: 1,
+                    column: 3,
+                    endLine: 1,
+                    endColumn: 37
```
Comment (btmills) [inline tests/lib/rules/consistent-return.js:262]: Reporting the entire function including its body seems inconsistent here. Everywhere else we have missing returns, the rule selects the function name, `function` keyword, or `=>` token. I'd expect this case to stop at the end of the `function` keyword like the rest.

**Review by btmills — CHANGES_REQUESTED** [review]
Thanks for taking on the rest of these end location reports! I see one test where the end location doesn't match my expectation. The start location matches what I'd expect, so this looks like the rule needs a new special case where it was previously able to use a fallback when it reported just start locations.

**Review by nzakas — APPROVED** [review]
LGTM

File: tests/lib/rules/consistent-return.js (hunk 2 above)
Comment (snitin315) [reply tests/lib/rules/consistent-return.js:262]: Hmm, should stop at tge end of `function`, I will update 👍

File: tests/lib/rules/consistent-return.js (hunk 1 above)
Comment (snitin315) [reply tests/lib/rules/consistent-return.js:320]: Makes sense, I will add these.

**Review by btmills — APPROVED** [review]
LGTM. Thanks @snitin315!

**Pull request merged**